test/jdk/java/awt/print/PrinterJob/PrintTextTest.java

```diff
@@ -21,193 +21,212 @@
-        p.add(name, ptt);
-        book.append(ptt, portrait);
-        book.append(ptt, landscape);
+        name = "Page " + page++;
+        pt = new PrintText(name, yfont, null, false, preferredSize);
+        pane.add(pt, BorderLayout.CENTER);
+        pane.add(name, pt);
+        book.append(pt, portrait);
+        book.append(pt, landscape);
 
         if (System.getProperty("os.name").startsWith("Windows")) {
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
-            ptt = new PrintJAText(name, font, null, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            name = "Page " + page++;
+            pt = new PrintJapaneseText(name, font, null, true, preferredSize);
+            pane.add(pt, BorderLayout.CENTER);
+            pane.add(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
 
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
+            name = "Page " + page++;
             rotTx = AffineTransform.getRotateInstance(0.15);
-            ptt = new PrintJAText(name, font, rotTx, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            pt = new PrintJapaneseText(name, font, rotTx, true, preferredSize);
+            pane.add(pt, BorderLayout.CENTER);
+            pane.add(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
         }
 
         pjob.setPageable(book);
 
-        JFrame f = new JFrame();
-        f.add(BorderLayout.CENTER, p);
-        f.addWindowListener(new WindowAdapter() {
-            public void windowClosing(WindowEvent e) {System.exit(0);}
+        JButton printButton = new JButton("Print");
+        printButton.addActionListener(event -> {
+            try {
+                if (pjob.printDialog()) {
+                    pjob.print();
+                }
+            } catch (PrinterException e) {
+                throw new RuntimeException(e.getMessage());
+            }
         });
+
+        JFrame f = new JFrame("PrintTextTest");
+        f.add(BorderLayout.CENTER, pane);
+        f.add(BorderLayout.SOUTH, printButton);
         f.pack();
-        f.show();
 
-        try {
-            if (pjob.printDialog()) {
-                pjob.print();
-            }
-        } catch (PrinterException e) {
-            throw new RuntimeException(e.getMessage());
-        }
+        PassFailJFrame.builder()
+            .title("PrintTextTest")
+            .instructions(INSTRUCTIONS)
+            .columns(60)
+            .testUI(f)
+            .build()
+            .awaitAndCheck();
     }
 
     // The test needs a physical font that supports Latin.
```

Comment:
The `getPhysicalFont` could be updated so that its `switch` statement uses `Font.` constants, I'm pretty sure that was the intention.

Reply:
It looks like it's trying to be extra resilient by switching on `n.toLowerCase()` (i.e. case-insensitive). Is it OK to make it case-sensitive and use the constants?

Reply:
I thought about it too. Using constants and case-sensitive switch should be enough.

Also, I think the `switch` statement should have `continue` rather than `break` for these logical fonts… unless the logical fonts are guaranteed to be in the end of the array returned by `GraphicsEnvironment.getAvailableFontFamilyNames()`.

Reply:
Done!